packages/example/src/components/SignMessage/SignMessage.tsx

```diff
@@ -1,17 +1,44 @@
-import React from "react";
+import React, {useState} from "react";
 import {Button, TextField, Card, CardContent, CardHeader, Grid, Box} from '@material-ui/core/';
+import {pluginOrigin, getInjectedMetamaskExtension} from "../../services/metamask";
 
 export const SignMessage = () => {
+    const [textFieldValue, setTextFieldValue] = useState<string>();
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setTextFieldValue(event.target.value);
+      };
+
+    const onSubmit = async () => {
+        if(textFieldValue) {
+            const provider = await getInjectedMetamaskExtension();
+            
+            if(provider && provider.signer && provider.signer.signRaw) {
+
+                const metamaskSnapApi = await provider.getMetamaskSnapApi();
+                const address = await metamaskSnapApi.getAddress(pluginOrigin);
+
+                const messageSignResponse = await provider.signer.signRaw({
+                    data: textFieldValue,
```

Comment:
You didn't hex encode value

Reply:
You can use this from @polkadot/util:
https://github.com/polkadot-js/common/blob/master/packages/util/src/string/toHex.ts